index/store/boltdb/writer.go

```diff
@@ -38,7 +38,8 @@ func (w *Writer) ExecuteBatch(batch store.KVBatch) error {
 	if err != nil {
 		return err
 	}
-
+	defer tx.Rollback()
```

Comment:
@daschl unfortunately it depends.  I generally follow the pattern where a constructor for a Closable thing returns nil, and not a half-open thing, for the case where an error occurs while creating/opening.  Thus the case of Rolling back the thing you *just* failed to open isn't required (because we did any necessary clean up for you).  So, for the me contract is, you **MUST** call Close() on anything that was successfully opened.  And if it was not successfully opened in the first place, that isn't required.

But, in the end you have to check, because not everyone follows that.